src/scripts/breeding/HatcheryHelper.ts

```diff
@@ -188,7 +188,7 @@ class HatcheryHelpers {
             egg.addSteps(steps, multiplier);
 
             // Check if the egg is ready to hatch
-            if (egg.progress() >= 100 || egg.isNone()) {
+            if (egg.canHatch()) {
```

Comment:
We should move the code for checking if this slot is empty below this block maybe,
otherwise if the filter is ever empty the helpers won't continue breeding.

Reply:
I can't see that code anywhere?

Reply:
You removed the check for ` || egg.isNone()`,
Which tris to hatch the egg (which shouldn't do anything),
Then replaces the egg.

Reply:
egg.canHatch() has the same check, right?

Reply:
`egg.canHatch()` does the opposite, checks that the egg slot isn't empty.